# Introdução à Ciência de Dados - UFPB
Professor: Yuri Malheiros

## Projeto

### Dataset

In [26]:
import pandas as pd
import numpy as np
df = pd.read_csv("./datasets/LED.csv")
df


Country  Year      Status  Life expectancy   Adult Mortality  \
0     Afghanistan  2015  Developing              65.0            263.0   
1     Afghanistan  2014  Developing              59.9            271.0   
2     Afghanistan  2013  Developing              59.9            268.0   
3     Afghanistan  2012  Developing              59.5            272.0   
4     Afghanistan  2011  Developing              59.2            275.0   
...           ...   ...         ...               ...              ...   
2933     Zimbabwe  2004  Developing              44.3            723.0   
2934     Zimbabwe  2003  Developing              44.5            715.0   
2935     Zimbabwe  2002  Developing              44.8             73.0   
2936     Zimbabwe  2001  Developing              45.3            686.0   
2937     Zimbabwe  2000  Developing              46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   \
0                62     0.01               71.279624         65.0      1154   
1                64     0.01               73.523582         62.0       492   
2                66     0.01               73.219243         64.0       430   
3                69     0.01               78.184215         67.0      2787   
4                71     0.01                7.097109         68.0      3013   
...             ...      ...                     ...          ...       ...   
2933             27     4.36                0.000000         68.0        31   
2934             26     4.06                0.000000          7.0       998   
2935             25     4.43                0.000000         73.0       304   
2936             25     1.72                0.000000         76.0       529   
2937             24     1.68                0.000000         79.0      1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
0     ...    6.0               8.16         65.0        0.1  584.259210   
1     ...   58.0               8.18         62.0        0.1  612.696514   
2     ...   62.0               8.13         64.0        0.1  631.744976   
3     ...   67.0               8.52         67.0        0.1  669.959000   
4     ...   68.0               7.87         68.0        0.1   63.537231   
...   ...    ...                ...          ...        ...         ...   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358878   

      Population   thinness  1-19 years   thinness 5-9 years  \
0     33736494.0                   17.2                 17.3   
1       327582.0                   17.5                 17.5   
2     31731688.0                   17.7                 17.7   
3      3696958.0                   17.9                 18.0   
4      2978599.0                   18.2                 18.2   
...          ...                    ...                  ...   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
0                               0.479       10.1  
1                               0.476       10.0  
2                               0.470        9.9  
3                               0.463        9.8  
4                               0.454        9.5  
...                               ...        ...  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427      

### Limpando e corrigindo colunas

In [79]:
oldcols = list(df.columns)
newcols = []
for col in oldcols:
    newcols.append(col.strip().replace('  ', ' ').replace(' ', '_').lower())
df.columns = newcols
df.rename(columns={'thinness_1-19_years':'thinness_10-19_years'}, inplace=True)

### Lidar com valores inválidos/suspeitos

In [18]:
df.describe().iloc[0:,1:]

Life expectancy   Adult Mortality  infant deaths      Alcohol  \
count       2928.000000      2928.000000    2938.000000  2744.000000   
mean          69.224932       164.796448      30.303948     4.602861   
std            9.523867       124.292079     117.926501     4.052413   
min           36.300000         1.000000       0.000000     0.010000   
25%           63.100000        74.000000       0.000000     0.877500   
50%           72.100000       144.000000       3.000000     3.755000   
75%           75.700000       228.000000      22.000000     7.702500   
max           89.000000       723.000000    1800.000000    17.870000   

       percentage expenditure  Hepatitis B       Measles          BMI   \
count             2938.000000  2385.000000    2938.000000  2904.000000   
mean               738.251295    80.940461    2419.592240    38.321247   
std               1987.914858    25.070016   11467.272489    20.044034   
min                  0.000000     1.000000       0.000000     1.000000   
25%                  4.685343    77.000000       0.000000    19.300000   
50%                 64.912906    92.000000      17.000000    43.500000   
75%                441.534144    97.000000     360.250000    56.200000   
max              19479.911610    99.000000  212183.000000    87.300000   

       under-five deaths         Polio  Total expenditure  Diphtheria   \
count         2938.000000  2919.000000         2712.00000  2919.000000   
mean            42.035739    82.550188            5.93819    82.324084   
std            160.445548    23.428046            2.49832    23.716912   
min              0.000000     3.000000            0.37000     2.000000   
25%              0.000000    78.000000            4.26000    78.000000   
50%              4.000000    93.000000            5.75500    93.000000   
75%             28.000000    97.000000            7.49250    97.000000   
max           2500.000000    99.000000           17.60000    99.000000   

          HIV/AIDS            GDP    Population   thinness  1-19 years  \
count  2938.000000    2490.000000  2.286000e+03            2904.000000   
mean      1.742103    7483.158469  1.275338e+07               4.839704   
std       5.077785   14270.169342  6.101210e+07               4.420195   
min       0.100000       1.681350  3.400000e+01               0.100000   
25%       0.100000     463.935626  1.957932e+05               1.600000   
50%       0.100000    1766.947595  1.386542e+06               3.300000   
75%       0.800000    5910.806335  7.420359e+06               7.200000   
max      50.600000  119172.741800  1.293859e+09              27.700000   

        thinness 5-9 years  Income composition of resources    Schooling  
count          2904.000000                      2771.000000  2775.000000  
mean              4.870317                         0.627551    11.992793  
std               4.508882                         0.210904     3.358920  
min               0.100000                         0.000000     0.000000  
25%               1.500000                         0.493000    10.100000  
50%               3.300000                         0.677000    12.300000  
75%               7.200000                         0.779000    14.300000  
max              28.600000                         0.948000    20.700000

min infant_deaths=0; 
min adult_mortality=1; 
min under-five_deaths=0;
max bmi=87.3;

In [82]:
df.infant_deaths = df.infant_deaths.replace(0, np.nan)
df['under-five_deaths'] = df['under-five_deaths'].replace(0, np.nan)
#adult_mortality
#bmi

### Nulos

In [83]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2938 entries, 0 to 2937
Data columns (total 22 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   country                          2938 non-null   object 
 1   year                             2938 non-null   int64  
 2   status                           2938 non-null   object 
 3   life_expectancy                  2928 non-null   float64
 4   adult_mortality                  2928 non-null   float64
 5   infant_deaths                    2090 non-null   float64
 6   alcohol                          2744 non-null   float64
 7   percentage_expenditure           2938 non-null   float64
 8   hepatitis_b                      2385 non-null   float64
 9   measles                          2938 non-null   int64  
 10  bmi                              2904 non-null   float64
 11  under-five_deaths                2153 non-null   float64
 12  polio               

### Heatmap